state-sync/state-sync-v2/data-streaming-service/src/data_stream.rs

```diff
@@ -95,22 +84,27 @@ pub struct DataStream<T> {
 
 impl<T: DiemDataClient + Send + Clone + 'static> DataStream<T> {
     pub fn new(
+        config: DataStreamingServiceConfig,
         data_stream_id: DataStreamId,
         stream_request: &StreamRequest,
         diem_data_client: T,
         notification_id_generator: Arc<U64IdGenerator>,
         advertised_data: &AdvertisedData,
     ) -> Result<(Self, DataStreamListener), Error> {
         // Create a new data stream listener
-        let (notification_sender, notification_receiver) =
-            diem_channel::new(QueueStyle::KLAST, DATA_STREAM_CHANNEL_SIZE, None);
+        let (notification_sender, notification_receiver) = diem_channel::new(
+            QueueStyle::KLAST,
+            config.max_data_stream_channel_sizes as usize,
```

Comment:
You can probably just make this value `usize`.  Since we should never get close to even the high values as a config value.

Reply:
I believe that `usize` and `u64` are the same size on 64 bit machines. I'm thinking I'll just leave it as is because it's consistent. 😄